src/modules/command/cmd.rs

```diff
@@ -43,12 +46,34 @@ impl SyntaxModule<ParserMetadata> for Command {
             let tok = meta.get_current_token();
             (self.strings, self.interps) = parse_interpolated_region(meta, &InterpolatedRegionType::Command)?;
             
-            // Set position for both failed and succeeded handlers
+            // Set position for failed, succeeded, and then handlers
             let position = PositionInfo::from_between_tokens(meta, tok.clone(), meta.get_current_token());
             self.failed.set_position(position.clone());
-            self.succeeded.set_position(position);
+            self.succeeded.set_position(position.clone());
+            self.then.set_position(position);
             
-            // Try to parse succeeded block first
+            // Try to parse then block first
+            syntax(meta, &mut self.then)?;
+            
+            if self.then.is_parsed {
+                // Check for conflicts with failed or succeeded blocks
+                if token(meta, "failed").is_ok() {
+                    return error!(meta, meta.get_current_token() => {
+                        message: "Cannot use both 'then' and 'failed' blocks for the same command",
+                        comment: "Use either 'then' to handle both success and failure, or 'failed'/'succeeded' blocks, but not both"
+                    });
+                }
+                if token(meta, "succeeded").is_ok() {
+                    return error!(meta, meta.get_current_token() => {
+                        message: "Cannot use both 'then' and 'succeeded' blocks for the same command",
+                        comment: "Use either 'then' to handle both success and failure, or 'failed'/'succeeded' blocks, but not both"
+                    });
+                }
```

Comment:
Please use `if let Some(word) = token_by(|word| word == "failed" || word == "succeeded")` here to simplify this condition.